src/main/java/org/spongepowered/common/data/manipulator/mutable/entity/SpongeExperienceHolderData.java

```diff
@@ -0,0 +1,232 @@
+import org.spongepowered.common.data.manipulator.mutable.common.AbstractData;
+import org.spongepowered.common.data.value.mutable.SpongeBoundedValue;
+import org.spongepowered.common.util.GetterFunction;
+import org.spongepowered.common.util.SetterFunction;
+
+public class SpongeExperienceHolderData extends AbstractData<ExperienceHolderData, ImmutableExperienceHolderData> implements ExperienceHolderData {
+
+    private int level;
+    private int totalExp;
+    private int expSinceLevel;
+    private final int expBetweenLevels;
+
+    public SpongeExperienceHolderData(int level, int totalExp, int expSinceLevel, int expBetweenLevels) {
+        super(ExperienceHolderData.class);
+        this.level = level;
+        this.totalExp = totalExp;
+        this.expSinceLevel = expSinceLevel;
+        this.expBetweenLevels = expBetweenLevels;
+        registerGettersAndSetters();
+    }
+
+    public SpongeExperienceHolderData() {
+        this(0, 0, 0, 0);
+    }
+
+    @Override
+    public ExperienceHolderData copy() {
+        return new SpongeExperienceHolderData(this.level, this.totalExp, this.expSinceLevel, this.expBetweenLevels);
+    }
+
+    @Override
+    public ImmutableExperienceHolderData asImmutable() {
+        return new ImmutableSpongeExperienceHolderData(this.level, this.totalExp, this.expSinceLevel, this.expBetweenLevels);
+    }
+
+    @Override
+    public int compareTo(ExperienceHolderData o) {
+        return ComparisonChain.start()
+                .compare(o.level().get().intValue(), this.level)
+                .compare(o.totalExperience().get().intValue(), this.totalExp)
+                .compare(o.experienceSinceLevel().get().intValue(), this.expSinceLevel)
+                .result();
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        return new MemoryDataContainer()
+                .set(Keys.EXPERIENCE_LEVEL.getQuery(), this.level)
+                .set(Keys.TOTAL_EXPERIENCE.getQuery(), this.totalExp)
+                .set(Keys.EXPERIENCE_SINCE_LEVEL.getQuery(), this.expSinceLevel)
+                .set(Keys.EXPERIENCE_FROM_START_OF_LEVEL.getQuery(), this.expBetweenLevels);
+    }
+
+    @Override
+    public MutableBoundedValue<Integer> level() {
+        return new SpongeBoundedValue<Integer>(Keys.EXPERIENCE_LEVEL, 0, intComparator(), 0, Integer.MAX_VALUE, this.level);
+    }
+
+    @Override
+    public MutableBoundedValue<Integer> totalExperience() {
+        return new SpongeBoundedValue<Integer>(Keys.TOTAL_EXPERIENCE, 0, intComparator(), 0, Integer.MAX_VALUE, this.totalExp);
+    }
+
+    @Override
+    public MutableBoundedValue<Integer> experienceSinceLevel() {
+        return new SpongeBoundedValue<Integer>(Keys.EXPERIENCE_SINCE_LEVEL, 0, intComparator(), 0, Integer.MAX_VALUE, this.expSinceLevel);
+    }
+
+    // i am not sure about this, doesnt this have to be immutable??? you cannot
+    // set it
+    @Override
+    public MutableBoundedValue<Integer> getExperienceBetweenLevels() {
```

Comment:
Yes, it has to be immutable. You can make it an `ImmutableBoundedValue` since the contract is to have a `BoundedValue` and not a `MutableBoundedValue`.